tools/ci/src/main.rs

```diff
@@ -1,44 +1,99 @@
+        Some("format") => Check::FORMAT,
+        Some("clippy") => Check::CLIPPY,
+        Some("compile-fail") => Check::COMPILE_FAIL,
+        Some("test") => Check::TEST,
+        Some("doc-test") => Check::DOC_TEST,
+        Some("doc-check") => Check::DOC_CHECK,
+        Some("bench-check") => Check::BENCH_CHECK,
+        Some("example-check") => Check::EXAMPLE_CHECK,
+        Some("lints") => Check::FORMAT | Check::CLIPPY,
+        Some("doc") => Check::DOC_TEST | Check::DOC_CHECK,
+        Some("compile") => Check::COMPILE_FAIL | Check::BENCH_CHECK | Check::EXAMPLE_CHECK,
+        _ => Check::all(),
+    };
 
-    // See if clippy has any complaints.
-    // - Type complexity must be ignored because we use huge templates for queries
-    cmd!("cargo clippy --workspace --all-targets --all-features -- -D warnings -A clippy::type_complexity -W clippy::doc_markdown")
+    if what_to_run.contains(Check::FORMAT) {
+        // See if any code needs to be formatted
+        cmd!("cargo fmt --all -- --check")
+            .run()
+            .expect("Please run 'cargo fmt --all' to format your code.");
+    }
+
+    if what_to_run.contains(Check::CLIPPY) {
+        // See if clippy has any complaints.
+        // - Type complexity must be ignored because we use huge templates for queries
+        cmd!("cargo clippy --workspace --all-targets --all-features -- -D warnings -A clippy::type_complexity -W clippy::doc_markdown")
         .run()
         .expect("Please fix clippy errors in output above.");
+    }
 
-    // Run UI tests (they do not get executed with the workspace tests)
-    // - See crates/bevy_ecs_compile_fail_tests/README.md
-    {
+    if what_to_run.contains(Check::COMPILE_FAIL) {
+        // Run UI tests (they do not get executed with the workspace tests)
+        // - See crates/bevy_ecs_compile_fail_tests/README.md
         let _bevy_ecs_compile_fail_tests = pushd("crates/bevy_ecs_compile_fail_tests")
             .expect("Failed to navigate to the 'bevy_ecs_compile_fail_tests' crate");
-        cmd!("cargo test")
+        cmd!("cargo test --target-dir ../../target")
             .run()
             .expect("Compiler errors of the ECS compile fail tests seem to be different than expected! Check locally and compare rust versions.");
     }
 
-    // These tests are already run on the CI
-    // Using a double-negative here allows end-users to have a nicer experience
-    // as we can pass in the extra argument to the CI script
-    let args: Vec<String> = std::env::args().collect();
-    if args.get(1) != Some(&"nonlocal".to_string()) {
+    if what_to_run.contains(Check::TEST) {
         // Run tests
-        cmd!("cargo test --workspace")
+        cmd!("cargo test --workspace --lib --bins --tests --benches")
             .run()
             .expect("Please fix failing tests in output above.");
+    }
+
+    if what_to_run.contains(Check::DOC_TEST) {
+        // Run doc tests: these are ignored by `cargo test`
```

Comment:
This comment is incorrect. `cargo test` always includes doc-tests.
This is only true if you use the `--doc` flag, which (for whatever reason) cant be mixed with other Test Targets.

Id say only provide a single `cargo test --workspace` invokation.

Reply:
I don't remember why that comment was here before, fixed it. Taking the doc tests out of the common "test" step is useful to reduce CI duration as it doesn't need to run on all platforms.